presentation/src/main/java/org/sagebionetworks/research/presentation/model/implementations/ActiveUIStepViewBase.java

```diff
@@ -0,0 +1,126 @@
+
+import org.sagebionetworks.research.domain.step.interfaces.ActiveUIStep;
+import org.sagebionetworks.research.domain.step.interfaces.Step;
+import org.sagebionetworks.research.presentation.DisplayString;
+import org.sagebionetworks.research.presentation.model.ColorThemeView;
+import org.sagebionetworks.research.presentation.model.ImageThemeView;
+import org.sagebionetworks.research.presentation.model.UIActionView;
+import org.sagebionetworks.research.presentation.model.interfaces.ActiveUIStepView;
+import org.threeten.bp.Duration;
+
+public class ActiveUIStepViewBase extends UIStepViewBase implements ActiveUIStepView {
+    @NonNull
+    private final Duration duration;
+    private final boolean isBackgroundAudioRequired;
+
+    public ActiveUIStepViewBase(@NonNull final String identifier, final int navDirection,
+            @NonNull final ImmutableMap<String, UIActionView> actions,
+            @Nullable final DisplayString title,
+            @Nullable final DisplayString text,
+            @Nullable final DisplayString detail,
+            @Nullable final DisplayString footnote,
+            @Nullable final ColorThemeView colorTheme,
+            @Nullable final ImageThemeView imageTheme, @NonNull final Duration duration,
+            final boolean isBackgroundAudioRequired) {
+        super(identifier, navDirection, actions, title, text, detail, footnote, colorTheme, imageTheme);
+        this.duration = duration;
+        this.isBackgroundAudioRequired = isBackgroundAudioRequired;
+    }
+
+    public static ActiveUIStepViewBase fromActiveUIStep(@NonNull Step step) {
+        if (!(step instanceof ActiveUIStep)) {
+            throw new IllegalArgumentException("Provided step: " + step + " is not an ActiveUIStep.");
+        }
+
+        ActiveUIStep activeUIStep = (ActiveUIStep)step;
+        UIStepViewBase uiStepView = UIStepViewBase.fromUIStep(activeUIStep);
+        // The duration from the ActiveUIStep is a Double in seconds, we convert this to a long of milliseconds and
+        // round off any extra precision.
+        Double millis = (activeUIStep.getDuration() != null ? activeUIStep.getDuration() : 0D) * 1000;
+        Duration duration = Duration.ofMillis(millis.longValue());
+        return new ActiveUIStepViewBase(uiStepView.getIdentifier(), uiStepView.getNavDirection(),
+                uiStepView.getActions(), uiStepView.getTitle(), uiStepView.getText(), uiStepView.getDetail(),
+                uiStepView.getFootnote(), uiStepView.getColorTheme(), uiStepView.getImageTheme(), duration,
+                activeUIStep.isBackgroundAudioRequired());
+    }
+
+    @Override
+    @NonNull
+    public Duration getDuration() {
+        return duration;
+    }
+
+    @Override
+    public boolean isBackgroundAudioRequired() {
+        return isBackgroundAudioRequired;
+    }
+
+    @Override
+    public int describeContents() {
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel dest, int flags) {
```

Comment:
Do we have a test which covers this?

Reply:
I don't believe we do I will go ahead and write some more tests for the step views.